test/end2end_test.go

```diff
@@ -2433,7 +2433,8 @@ func testMetadataStreamingRPC(t *testing.T, e env) {
 
 			payload, err := newPayload(testpb.PayloadType_COMPRESSABLE, int32(reqSizes[index]))
 			if err != nil {
-				t.Fatal(err)
+				t.Error(err)
+				return
```

Comment:
This is where I meant to leave that comment.

Maybe defer stream.CloseSend() at the top of the function?